stdlib/public/runtime/ErrorObject.mm

```diff
@@ -371,15 +373,17 @@ static id getEmptyNSDictionary() {
   this->hashableBaseType.compare_exchange_strong(
       expectedType, hashableBaseType ? hashableBaseType
                                      : reinterpret_cast<const Metadata *>(1),
-      std::memory_order_acq_rel);
+      std::memory_order_relaxed, std::memory_order_relaxed);
```

Comment:
I don't know this part of the runtime logic but I would be dubious of the fact that this acq_rel or the other acquire barriers in this file are spurious.

> Furthermore, memory_order_acq_rel is not necessary for most of these operations anyway, as the value being assigned to the atomic variable does not depend on the current exact value of  said atomic variable.

That is not the right way to think about this - you should be reasoning about whether there are *other* memory updates apart from the atomic, that need to be made visible to the thread in this operation.